File: 492. Construct the Rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        l = int(math.sqrt(area))
        while l <= area:
            if area % l == 0 and l >= (area // l):
                return [l,area//l]
            l += 1
        return [area,1]
        

File: 492. Construct the Rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        // 获得面积的平方根
        // 从平方根开始向下取整，如果能被整除，则输出 int pair
        for (int l = (int)Math.sqrt(area); l <= area; l++) {
            if (area % l == 0 && l >= (area/l)) {
                return new int[]{l,area/l};
            }
        }
        return new int[]{area,1};
    }
}
